### PR TITLE
Shared constants between conmon and crio

### LINKED ISSUE #1060: Share constants between C and Go code
We have some duplicated constants between Go code and conmon that we should move to a header file.

### PR DESCRIPTION
**Pull request by fntlnz**
Signed-off-by: Lorenzo Fontana <lo@linux.com>

closes #1060

**- What I did**
Shared some common constants between crio and conmon.

**- How I did it**
Grep for the common patterns I identified and then wrote a `config.h` file that has been imported where needed.

**- How to verify it**

**- Description for the changelog**
Shared constants between conmon and crio

### CODE REVIEW

**Comment by fntlnz** [comment]
@mrunalp I'm waiting for @rhatdan to reply [this comment](https://github.com/kubernetes-incubator/cri-o/pull/1052/files/e66da6046d0e8eef1d5eeb949579ba6dd843f88c#r151841747) to know if we need to share `/var/run/crio` too.
If you think that there's something else please let me know.

**Comment by fntlnz** [comment]
I used relative import paths for headers, if needed I can edit the include path flags to just use `conmon/config.h`

**Comment by TomSweeneyRedHat** [comment]
@fntlnz thanks for the PR!  I've kicked off the tests, the code LGTM but @mrunalp and/or @runcom will need to review further.

File: cmd/crio/main.go [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package main
 
+// #include "../../conmon/config.h"
```
Comment (runcom) [inline cmd/crio/main.go]: I'm not sure if it's ok to pull in C code for just one constant....

**Comment by runcom** [comment]
I'm not happy with this guys, maybe defaults must be passed down to C code directly after being defined in golang rather than doing this. Actually, I'd love this patch to be: define common consts in golang, and pass them down when needed to C.
This way isn't just pretty nor functional

**Comment by fntlnz** [comment]
@runcom while I agree with you that having this done without binding is a good alternative, I think that’s worth asking @rhatdan since I mostly followed [his suggestion](https://github.com/kubernetes-incubator/cri-o/pull/1052#issuecomment-339033870)

**Comment by runcom** [comment]
Well yeah, that's one of the possible approach, and looking at this PR it's not the one I prefer :P

**Comment by rhatdan** [comment]
I think we need @mrunalp to make the decision on this one.
There are other constants that we might want to share 
#define DEFAULT_SOCKET_PATH "/var/lib/crio"

**Comment by rhatdan** [comment]
@runcom So you would define a constant in a small go program and create the include.h file from it? Then you would not need to use golang C Bindings?

**Comment by rhatdan** [comment]
Ok this PR Has gotten rather old.  Lets follow @runcom suggestion.  Lets create a go program that outpus an inlcude file, that conmon can then consume.

**Comment by rhatdan** [comment]
I am replacing this PR with my own that generates the config.h from the go code.
https://github.com/kubernetes-incubator/cri-o/pull/1414

**Pull request closed without merging**